src/pushjack/apns.py

```diff
@@ -764,6 +776,346 @@ class APNSExpiredToken(namedtuple('APNSExpiredToken', ['token', 'timestamp'])):
+                offset from now. Defaults to ``None`` which uses
+                :attr:`default_expiration_offset`.
+            low_priority (boolean, optional): Whether to send notification with
+                the low priority flag. Defaults to ``False``.
+            max_payload_length (int, optional): The maximum length of the
+                payload to send. Message will be trimmed if the size is
+                exceeded. Use 0 to turn off. Defaults to ``None`` which uses
+                attr:`default_max_payload_length`.
+
+        Keyword Args:
+            badge (int, optional): Badge number count for alert. Defaults to
+                ``None``.
+            sound (str, optional): Name of the sound file to play for alert.
+                Defaults to ``None``.
+            category (str, optional): Name of category. Defaults to ``None``.
+            content_available (bool, optional): If ``True``, indicate that new
+                content is available. Defaults to ``None``.
+            title (str, optional): Alert title.
+            title_loc_key (str, optional): The key to a title string in the
+                ``Localizable.strings`` file for the current localization.
+            title_loc_args (list, optional): List of string values to appear in
+                place of the format specifiers in `title_loc_key`.
+            action_loc_key (str, optional): Display an alert that includes the
+                ``Close`` and ``View`` buttons. The string is used as a key to
+                get a localized string in the current localization to use for
+                the right button’s title instead of ``"View"``.
+            loc_key (str, optional): A key to an alert-message string in a
+                ``Localizable.strings`` file for the current localization.
+            loc_args (list, optional): List of string values to appear in place
+                of the format specifiers in ``loc_key``.
+            launch_image (str, optional): The filename of an image file in the
+                app bundle; it may include the extension or omit it.
+            mutable_content (bool, optional): if ``True``, triggers Apple
+                Notification Service Extension. Defaults to ``None``.
+            thread_id (str, optional): Identifier for grouping notifications.
+                iOS groups notifications with the same thread identifier
+                together in Notification Center. Defaults to ``None``.
+            extra (dict, optional): Extra data to include with the alert.
+        """
+        assert device_id, 'Must provide device IDs.'
+
+        if max_payload_length is None:
+            max_payload_length = self.default_max_payload_length
+
+        message = APNSMessage(message,
+                              max_payload_length=max_payload_length,
+                              **options)
+
+        validate_tokens([device_id])
+        validate_message(message)
```

Comment:
This will validate the message size against the binary protocol. I think the HTTP/2 API has a higher limit (4K vs 2K).

Reply:
Thanks. I overlooked this bit. I will address it.